src/server/svr_func.c

```diff
@@ -5102,6 +5103,28 @@ check_and_run_jobs(struct prov_vnode_info * prov_vnode_info)
 
 	rc = is_runnable(pjob, prov_vnode_info);
 	if (rc == 0) {
+		if (do_sync_mom_hookfiles || sync_mom_hookfiles_proc_running) {
+
+			/**
+			 * If mom hook files sync is in process then create
+			 * a time task where you perform this check again,
+			 * and start the job once it is done
+			 */
+
+			DBPRT(("%s: setting the time task as sync mom"
+				"hookfiles is not completed\n"
+				, __func__))
+
+			/* set a work task to run after 5 sec from now */ 
+			if (!set_task(WORK_Timed, time_now + 5, 
+				is_hooks_sync_done, pjob)){
+				log_err(errno, __func__,
+					"Unable to set task for is_hooks_sync_done; requeuing the job");
+				(void)force_reque(pjob);
+			}
+			return;
```

Comment:
Hi Arun,
>  The scenario where provisioning is still in progress for some nodes (-1 case), isn't that a bug then because in that case some nodes are still being provisioned and we remove entry from tracking table?

We remove the provisioning record only for the nodes which have finished provisioning. Other nodes which are still in provisioning will have their records intact, and will be deleted once they finish provisioning or fail provisioning .